src/components/MapView/Analyser/index.tsx

```diff
@@ -187,8 +195,47 @@ function Analyser({ classes }: AnalyserProps) {
     };
 
     await dispatch(requestAndStoreAnalysis(params));
+    // eslint-disable-next-line fp/no-mutating-methods
+    history.replace(
+      `?hazardLayerId=${hazardLayerId}&baselineLayerId=${baselineLayerId}&date=${selectedDate}&statistic=${statistic}&aboveThreshold=${aboveThreshold}&belowThreshold=${belowThreshold}`,
+    );
   };
 
+  useEffect(() => {
+    const {
+      hazardLayerParamId,
+      baselineLayerParamId,
+      selectedParamDate,
+      statisticParam,
+      aboveThresholdParam,
+      belowThresholdParam,
+    } = extractPropsFromURL(history.location.search);
+
+    if (!hazardLayerParamId || !baselineLayerParamId || !statisticParam) {
+      return;
+    }
+
+    if (isLayerKey(hazardLayerParamId)) {
+      setHazardLayerId(hazardLayerParamId);
+    }
+    if (isLayerKey(baselineLayerParamId)) {
+      setBaselineLayerId(baselineLayerParamId);
+    }
+    if (isStatistic(statisticParam)) {
+      setStatistic(statisticParam);
+    }
+
+    setSelectedDate(selectedParamDate);
+    setAboveThreshold(aboveThresholdParam);
+    setBelowThreshold(belowThresholdParam);
+
+    // Avoid Running Analyser if boundary Layer data is Null
+    if (hazardLayerId && baselineLayerId && selectedDate) {
```

Comment:
I'd prefer if we'd use a try/catch so we can reuse the existing verification inside `runAnalyser()`

Reply:
That will always return throw an error. The point of `if` statement is to avoid executing function at all. Before the values are updated, it tries to execute `runAnalyser` (and the layers are not defined yet).